active/InformationBar/InformationBar.md

```diff
@@ -0,0 +1,622 @@
+| InfoBarActionButtonCornerRadius | Action button corner radius
+| InfoBarHyperlinkButtonFontSize | Hyperlink button font size
+| InfoBarHyperlinkButtonHeight | Hyperlink button height
+| InfoBarHyperlinkButtonPadding | Hyperlink button padding thickness
+| InfoBarCloseButtonSymbol | Close button symbol glyph
+| InfoBarCloseButtonStyle | Close button style
+
+
+# Appendix
+
+## Design References
+UI Elements for InfoBar
+
+
+![Toolkit page for InfoBar UI elements](images/Docked_Toolkit.png)
+
+### Visual Components
+| Component |  Notes |
+|:---:|:---|
+| Container | - We recommend to place InfoBars in a layout control where the control can expand horizontally to the width of the content area.
+| Icon | - Defined by either the Severity or by IconSource <br> - Can be removed by setting IsIconVisible to false
+| Title | - Semi-bolded and appears right of the Icon <br> - Recommended to be 50 characters or less
+| Message | - Will appear to the right of the Title in single-height notifications, otherwise will be on a new line <br> - Recommended to be 512 characters or less
+| Action button |  - Will appear directly to the right of the Message in single height notifications, otherwise will be on its own new line. <br> - An action button can be any class that inherits ButtonBase however Button, HyperlinkButton, and Checkbox are our recommended options <br> - Additional action buttons may be added through custom XAML content
+| Close button | - Will appear as 'X' by default <br> - Can be removed by setting IsUserDismissable to false
+| Content | - Can be customizable to include text, hyperlinks, and any other XAML content <br> - Appears in its own line underneath the other InfoBar content
+
+## Behavioral Components
+| Property | Notes |
+|:---:|:---|
+| Opening | - An info bar is shown by setting its IsOpen property to true. |
+| Closing | There are two ways an info bar can close: <br>- The program sets the IsOpen property to false <br> - The user invokes the Close button. <br> Use the InfoBarCloseReason to determine which case has occurred. <br> Closing can be prevented by setting the Cancel property to true. You can use a deferral to respond asynchronously to the event. |
+
+### Canceling and deferring close
+The Closing event can be used to cancel and/or defer the close of an InfoBar. 
+This can be used to keep the InfoBar open or allow time for a custom action to occur. 
+When the closing of an InfoBar is canceled, IsOpen will go back to true, however, 
+it will stay false during the deferral. A programmatic close can also be canceled.
+
+
+XAML
+```xml
+<controls:InfoBar x:Name="UpdateAvailable"
+    Title="Update Available"
+    Message="Please close this tip to apply required security updates to this application"
+    Closing="InfoBar_Closing">
+</controls:InfoBar>
+```
+C#
+```C#
+public void InfoBar_Closing(InfoBar sender, InfoBarClosingEventArgs args)
+{
+    if (args.Reason == InfoBarCloseReason.CloseButton) {
+        // if scenario failed
```

Comment:
Use of comment here is confusing. This isn't commenting the code; it's representing code that is missing. Better would be to call a function not implemented here.

```c#
if (args.Reason == InfoBarCloseReason.CloseButton) {
  if (!ApplyUpdates()) {
    // could not apply updates - do not close
    args.Cancel = true;
  }
}
```